pkg/virt-handler/migration-target.go

```diff
@@ -0,0 +1,853 @@
+			vmi.Status.VolumeStatus[i].PersistentVolumeClaimInfo = status
+		}
+	}
+}
+
+func (c *MigrationTargetController) syncVolumes(vmi *v1.VirtualMachineInstance) error {
+	// The VolumeStatus is used to retrive additional information for the volume handling.
+	// For example, for filesystem PVC, the information are used to create a right size image.
+	// In the case of migrated volumes, we need to replace the original volume information with the
+	// destination volume properties.
+	replaceMigratedVolumesStatus(vmi)
+	err := hostdisk.ReplacePVCByHostDisk(vmi)
+	if err != nil {
+		return err
+	}
+
+	// give containerDisks some time to become ready before throwing errors on retries
+	info := c.launcherClients.GetLauncherClientInfo(vmi)
+	if ready, err := c.containerDiskMounter.ContainerDisksReady(vmi, info.NotInitializedSince); !ready {
+		if err != nil {
+			return err
+		}
+		c.queue.AddAfter(controller.VirtualMachineInstanceKey(vmi), time.Second*1)
+		return container_disk.ErrWaitingForDisks
+	}
+
+	// Mount container disks
+	err = c.containerDiskMounter.MountAndVerify(vmi)
+	if err != nil {
+		return err
+	}
+
+	// Mount hotplug disks
+	if attachmentPodUID := vmi.Status.MigrationState.TargetAttachmentPodUID; attachmentPodUID != types.UID("") {
+		cgroupManager, err := getCgroupManager(vmi, c.host)
+		if err != nil {
+			return err
+		}
+		if err := c.hotplugVolumeMounter.MountFromPod(vmi, attachmentPodUID, cgroupManager); err != nil {
+			return fmt.Errorf("failed to mount hotplug volumes: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *MigrationTargetController) processVMI(vmi *v1.VirtualMachineInstance) error {
+	isUnresponsive, isInitialized, err := c.launcherClients.IsLauncherClientUnresponsive(vmi)
+	if err != nil {
+		return err
+	}
+
+	if !isInitialized {
+		log.Log.Object(vmi).V(4).Info("launcher client is not initialized")
+		c.queue.AddAfter(controller.VirtualMachineInstanceKey(vmi), time.Second*1)
+		return nil
+	} else if isUnresponsive {
+		return goerror.New(fmt.Sprintf("Can not update a VirtualMachineInstance with unresponsive command server."))
+	}
+
+	if migrationNeedsFinalization(vmi.Status.MigrationState) {
+		log.Log.Object(vmi).V(4).Info("finalize migration")
+		c.finalizeMigration(vmi)
+		return nil
+	}
+
+	client, err := c.launcherClients.GetLauncherClient(vmi)
+	if err != nil {
+		return fmt.Errorf(unableCreateVirtLauncherConnectionFmt, err)
+	}
+
+	if migrations.MigrationFailed(vmi) {
+		// if the migration failed, signal the target pod it's okay to exit
+		err = client.SignalTargetPodCleanup(vmi)
+		if err != nil {
+			return err
+		}
+		log.Log.Object(vmi).Infof("Signaled target pod for failed migration to clean up")
+		// nothing left to do here if the migration failed.
+		// Re-enqueue to trigger handler final cleanup
+		c.queue.AddAfter(controller.VirtualMachineInstanceKey(vmi), time.Second)
+		delete(vmi.Labels, v1.MigrationTargetNodeNameLabel)
```

Comment:
```suggestion
```
This is premature, my bad, the other delete in `finalizeMigration()` will take care of both the failed and successful cases. (deleting the label and re-enqueuing doesn't make sense anyway, since without the label the VMI becomes invisible)